BookShelfBusinessLogic/LibraryService.cs

```diff
@@ -0,0 +1,379 @@
+{
+    /// <summary>
+    /// Class containing all methods for data exchange with the database.
+    /// Accepts context - either external DB or in memory.
+    /// </summary>
+    public class LibraryService : ILibraryService
+    {
+        /// <summary>
+        /// Context to be working with.
+        /// Containes collections of type DbSet for all models.
+        /// </summary>
+        private ILibraryContext _db;
+
+        public LibraryService()
+        {
+        }
+
+        /// <summary>
+        /// Initializes new instance of LibraryService.
+        /// </summary>
+        /// <param name="context">A data context</param>
+        public LibraryService(ILibraryContext context)
+        {
+            _db = context;
+        }
+
+        // #####################################
+        // ALL METHODS RELATED TO BOOK:
+        // #####################################
+        #region BOOKS
+
+        /// <summary>
+        /// Selects all instances of books from  data context collection
+        /// </summary>
+        /// <returns>The collection of books </returns>
+        public IEnumerable<Book> GetBooks()
+        {
+            return _db.Books;
+        }
+
+        /// <summary>
+        /// Looks for an instance of book with the specifies Id.
+        /// Creates a BookView instance for full representation (includes genres and authors of this book).
+        /// </summary>
+        /// <param name="id"> Id to look for </param>
+        /// <returns> An instance of BookView or null if not found </returns>
+        public BookView GetBookById(int id)
+        {
+            Book book =_db.Books.FirstOrDefault(b => b.Id == id);
+
+            if (book != null)
+            {
+                BookView bookView = new BookView(book, _db);
+
+                return bookView;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Looks for an instance of book with the specifies Id
+        /// and tries to set its properties equal to the book passed as argument, if found.
+        /// </summary>
+        /// <param name="id"> Id to look for</param>
+        /// <param name="book"> An instance of book  to be copied</param>
+        /// <returns> True if such book found and updated, false - if not found </returns>
+        public bool IsBookUpdated(int id, Book book)
+        {
+            Book bookToUpdate = _db.Books.First(b => b.Id == id);
+
+            if (bookToUpdate != null)
+            {
+                bookToUpdate.Title = book.Title;
+                _db.SaveChanges();
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Tries to add new book to the collection
+        /// </summary>
+        /// <param name="book"> An instace of book to be added </param>
+        /// <returns> True if added, false if such book already exists </returns>
+        public bool IsBookAdded(Book book)
```

Comment:
It`s bad naming, for this and all similar methods. If a method is called IsBookAdded I expect that it will only check if a book is there, but not add a book and return if it is there.